### DISCUSSION TITLE
how to make multi tables have a same parent

### DISCUSSION THREAD

**Discussion by BrightVan** [simolus3/drift#2509] [Q&A]
```dart
class Base extends Table{
 TextColumn get userId => text().references(User, #id)();

  TextColumn get id => text().clientDefault(() => Xid.string())();
 DateTimeColumn get insertTime => dateTime().withDefault(currentDateAndTime)();

  DateTimeColumn get updateTime => dateTime().withDefault(currentDateAndTime)();

  TextColumn get alias => text()();
 @override
  Set<Column> get primaryKey => {id};
}

class TableA extends Base{
   TextColumn get xxx => text()();
}

class TableB extends Base{
   TextColumn get xxx => text()();
}

class TableC extends Base{
   TextColumn get xxx => text()();
}

```
the question is that the generate code TableAData, TableBData, TableCData extends DataClass not BaseData. and  when i select from Table A,B,C, i have to write many dumplicate code, i can not  use Polymorphism

**Comment by simolus3** [comment]
There is no support for this in drift. A similar request is https://github.com/simolus3/drift/issues/881, but at the moment I have no plans for this feature. The only solution for this is to use [custom row classes](https://drift.simonbinder.eu/docs/advanced-features/custom_row_classes/) with the desired type hierarchy.
